cmd/default-evaluator/main.go

```diff
@@ -21,5 +21,5 @@ import (
 )
 
 func main() {
-	appmain.RunApplication("evaluator", evaluator.BindServiceFor(defaulteval.Evaluate))
+	appmain.RunApplication("evaluator", evaluator.BindServiceFor(evaluator.Evaluator{EvaluateFunc: defaulteval.Evaluate, Binders: defaulteval.Binders}))
```

Comment:
This pattern is awkward for a couple reasons:
- Passing the arguments inside a single a struct, instead of just using multiple args.
- The function and "binders" are always used together, so why doesn't that package do it?

Also, the change to evaluator breaks my testing change, I think that it's
reasonable to have a type for the function itself.

I think a cleaner pattern would be to invert how these are being used: instead
of the default evaluator being passed to the BindServiceFor, have the default
evaluator use BindServiceFor.

Within the defaulteval package, rename Evaluate to evaluate.  Add a BindService
function, which calls evaluator.BindServiceFor(evaluate)(p, b).  In addition, it
can bind any metrics. Then have this main function use that BindFunction instead
of BindServiceFor.